src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -172,6 +172,17 @@ def k8s_install_cli(client_version="1.4.5", install_location=None):
     except IOError as err:
         raise CLIError('Connection error while attempting to download client ({})'.format(err))
 
+def _validate_service_principal(sp_id):
+    from azure.cli.command_modules.role.custom import (
+        _graph_client_factory,
+        show_service_principal,
+    )
```

Comment:
Command modules should not have dependencies on each other...

Since command modules are/will be versioned independently, this could fail in the future if there's a change to the 'role' module that changes these methods (especially as the `_graph_client_factory()` is private).

Any logic needed should be included here.

You _can_ choose to leave it as is but there are no guarantees.